src/Stdio.php

```diff
@@ -249,9 +249,9 @@ private function restoreTtyMode()
             // remove dummy readline handler to turn to default input mode
             $this->usesExtReadlineHandler = false;
             readline_callback_handler_remove();
-        } elseif ($this->originalTtyMode !== null && $this->isTty()) {
+        } elseif ($this->originalTtyMode !== null && is_resource(STDIN) && $this->isTty()) {
             // Reset stty so it behaves normally again
-            shell_exec(sprintf('stty %s', $this->originalTtyMode));
+            shell_exec('stty ' . escapeshellarg($this->originalTtyMode));
```

Comment:
This simple change introduces a subtle bug fixed in #78. The `originalTtyMode` contains a trailing newline, passing this to `escapeshellarg()` means that this will not be passed as part of the argument to the `stty` bin which rejects this format because it does not expect a trailing newline.